### PR TITLE
feat: integrate midnight smoker in ci

### PR DESCRIPTION
**Pull request by couragecowardlydog**
Fix: https://github.com/pinojs/pino/issues/1570

@mcollina  @jsumners , please review and provide your comments 

Added a copy of example/basic.js in root as smoke.js

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,10 +23,11 @@
     "lint": "eslint .",
     "prepublishOnly": "tap --no-check-coverage test/internals/version.test.js",
     "test": "npm run lint && npm run transpile && tap --ts && jest test/jest && npm run test-types",
-    "test-ci": "npm run lint && npm run transpile && tap --ts --no-check-coverage --coverage-report=lcovonly && npm run test-types",
+    "test-ci": "npm run lint && npm run transpile && tap --ts --no-check-coverage --coverage-report=lcovonly && npm run test-types && npx midnight-smoker test-ci-smoke",
```
Comment (kibertoad) [inline package.json:26]: no additional dependency is needed?

File: package.json [hunk 2]
```diff
@@ -12,11 +12,9 @@
     "pino.d.ts",
     "bin.js",
     "browser.js",
-    "pretty.js",
-    "usage.txt",
     "test",
     "docs",
-    "example.js",
+    "smoke.js",
```
Comment (kibertoad) [inline package.json:17]: does it need to be included in npm package?

Comment (couragecowardlydog) [reply package.json:17]: Yes , midnight-smoker execute the script inside the package

File: package.json (hunk 1 above)
Comment (couragecowardlydog) [reply package.json:26]: Nope

Comment (mcollina) [reply package.json:26]: I would prefer to avoid `npx` and add `midnight-smoker` as a devdep.

Comment (couragecowardlydog) [reply package.json]: @mcollina, added `midnight-smoker` as dev-dep. Included separate script as well

**Review by mcollina — APPROVED** [review]
lgtm

**Review by kibertoad — APPROVED** [review]

**Comment by couragecowardlydog** [comment]
> lgtm

@mcollina, the pipeline broke for some of the case,  npm version compatibility 


```
Minimum Node.js versions supported: ^14.17.0 || ^16.13.0 || >=18.0.0
Minimum npm version supported: v7.0.0
```

**Comment by kibertoad** [comment]
@couragecowardlydog is it possible to run midnight smoker not through "exec" command?

**Comment by couragecowardlydog** [comment]
> @couragecowardlydog is it possible to run midnight smoker not through "exec" command?

@kibertoad, can I change the script like this 
```
"smoker": "node ./node_modules/.bin/smoker test-ci-smoke",
```

is that fine ?

**Comment by kibertoad** [comment]
I see no problem with that, if it ever stops working for whatever reason, CI will fail

**Comment by couragecowardlydog** [comment]
> I see no problem with that, if it ever stops working for whatever reason, CI will fail

Fixed that, can you please approve the workflow to see if CI goes through

File: package.json [hunk 3]
```diff
@@ -12,23 +12,23 @@
...
     "lib"
   ],
   "scripts": {
     "docs": "docsify serve",
     "browser-test": "airtap --local 8080 test/browser*test.js",
     "lint": "eslint .",
+    "smoker": "node ./node_modules/.bin/smoker test-ci-smoke",
```
Comment (mcollina) [inline package.json:24]:
```suggestion
    "smoker": "smoker test-ci-smoke",
```

this should work too

**Review by mcollina — CHANGES_REQUESTED** [review]

**Pull request closed without merging**